### ISSUE TITLE
Segfault when using quantifiers with QF_FP/QF_FPBV

### ISSUE THREAD

**Issue by 4tXJ7f** [Z3Prover/z3#544]
When using quantifiers with QF_FP/QF_FPBV, Z3 crashes with a segfault in `lackr::collect_terms()`:

```
$ cat out.txt
(assert (forall ((%rundef0 (_ FloatingPoint 11 53)))
  (fp.isNaN %rundef0)))
(apply qffp)
$ z3 -smt2 out.txt
Segmentation fault: 11
```

More discussion in #489.

**Comment by wintersteiger** [comment]
Regarding this issue, note that the tactic `qffp` is for QF_FP benchmarks, i.e., quantifier-free problems. Thus, the "correct" fix for this particular input is to throw an error.

**Comment by nunoplopes** [comment]
Christoph: allowing the use of QF_* tactics with quantifiers is desirable, and this has always worked.  For example, I do use the qfbv tactic regularly with quantifiers because it is faster than bv/ufbv tactics for certain benchmarks.

**Comment by wintersteiger** [comment]
@nunoplopes: Yes, and the current problem will also be solved in a way that allows this, but what you really want is a custom tactic. I general we can't assume that every tactic will work with every feature/theory.

**Comment by wintersteiger** [comment]
Should be fixed after #545
